src/parsing/tgraph.py

```diff
@@ -1,7 +1,10 @@
 from __future__ import annotations
+
+import json
```

Comment:
Adding `import` should be done on hand. Do not rely on your IDE to do it.
Read the `import` blocks twice to see where to add `import`.